### PR TITLE
refactor: query_tests - port a few OG to NG tests and remove many more that already ported

### PR DESCRIPTION
**Pull request by NGA-TRAN**
- [x] Remove OG tests that were ported (Thanks @crepererum )
 - [x] Rewrote a needed OG tests to NG tests
 - [x] Remove irrelevant OG tests
 - [ ] The only case I need more thinking before rewriting it in `NoData` and open #4488 for it

- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: query_tests/cases/in/no_stats_plans.expected [hunk 1]
```diff
@@ -1,13 +0,0 @@
--- Test Setup: OldTwoMeasurementsManyFieldsOneRubChunk
```
Comment (NGA-TRAN) [inline query_tests/cases/in/no_stats_plans.expected:1]: NG has no stats now. All query plans will be like this. No longer needs this irrelevant test

File: query_tests/cases/in/old_sql_system_tables.expected [hunk 2]
```diff
@@ -1,37 +0,0 @@
--- Test Setup: OldTwoMeasurementsManyFieldsOneRubChunk
```
Comment (NGA-TRAN) [inline query_tests/cases/in/old_sql_system_tables.expected:1]: We already have new system table tests. This is no irrelevant

File: query_tests/src/scenarios/delete.rs [hunk 3]
```diff
@@ -255,226 +251,76 @@ impl DbSetup for ThreeDeleteThreeChunks {
             )],
         };
 
-        // ----------------------
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/delete.rs:258]: This function includes both OG and NG scenarios. Remove the OG ones

File: query_tests/src/scenarios/delete.rs [hunk 4]
```diff
@@ -255,226 +251,76 @@ impl DbSetup for ThreeDeleteThreeChunks {
...
-        let lp = vec![
-            ChunkDataOld {
-                lp_lines: lp_lines_1.clone(),
-                chunk_stage: ChunkStageOld::Os,
+        // Scenarios
+        let scenarios = make_n_chunks_scenario_new(&[
+            ChunkDataNew {
+                lp_lines: lp_lines_1,
+                preds: preds.clone(),
+                // preds: vec![
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/delete.rs:275]: The original OG tests were meant to apply all 3 delete preds in every chunk. When we ported it to NG, we changed the original meaning. So this is a fix even though the test results are the same.

File: query_tests/src/scenarios/library.rs [hunk 5]
```diff
@@ -9,99 +9,19 @@ use db::{
...
-        all_scenarios_for_one_chunk, make_one_rub_or_parquet_chunk_scenario,
-        make_two_chunk_scenarios, ChunkStageNew,
-    },
+    util::{all_scenarios_for_one_chunk, make_two_chunk_scenarios, ChunkStageNew},
     DbScenario, DbSetup,
 };
 
-/// No data
-#[derive(Debug)]
-pub struct ChunkOrder {}
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/library.rs:28]: Remove this ChunkOrder setup because it was meant for OG life cycle which is no longer relevant in query_tests. This test is equivalent of NG 4 different tests in Router, Ingester, Compactor and Querier which have already been implemented.
I added a TODO below for end-to-end but it is kind a bit of different from this specific test

File: query_tests/cases/in/chunk_order.expected [hunk 6]
```diff
@@ -1,7 +0,0 @@
--- Test Setup: ChunkOrder
```
Comment (NGA-TRAN) [inline query_tests/cases/in/chunk_order.expected:1]: See below at `ChunkOrder` for the reason it is no longer needed

File: query_tests/src/scenarios/library.rs [hunk 7]
```diff
@@ -9,99 +9,19 @@ use db::{
...
-            scenario_name: "chunks where chunk ID alone cannot be used for ordering".into(),
-            db,
-        };
-
-        vec![scenario]
-    }
-}
+// TODO: this ChunkOrder tests OG chunks lifecyle which is very differnet in NG
+// keep this TODO here so in end-to-end tests, we will add very different life cycle tests
+// that inlcude verify catalog content and data
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/library.rs:24]: I am not sure the this comment is good for end-to-end tests either. Need to think a bit more before opening a ticket

File: query_tests/src/scenarios/library.rs [hunk 8]
```diff
@@ -598,75 +521,6 @@ impl DbSetup for TwoMeasurementsManyFieldsOneChunk {
     }
 }
 
-#[derive(Debug)]
-/// This has a single chunk for queries that check the state of the system
-///
-/// This scenario is OG-specific and can be used for `EXPLAIN` plans and system tables.
-pub struct OldTwoMeasurementsManyFieldsOneRubChunk {}
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/library.rs:605]: This RUB chunk is no longer relevant

File: query_tests/src/scenarios/library.rs [hunk 9]
```diff
@@ -598,75 +521,6 @@ impl DbSetup for TwoMeasurementsManyFieldsOneChunk {
...
-            db,
-        }]
-    }
-}
-
-#[derive(Debug)]
-/// This has two chunks for queries that check the state of the system
-///
-/// This scenario is OG-specific and can be used for `EXPLAIN` plans and system tables.
-pub struct OldTwoMeasurementsManyFieldsTwoChunks {}
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/library.rs:636]: It was ported

File: query_tests/src/scenarios/library.rs [hunk 10]
```diff
@@ -755,8 +609,6 @@ pub struct OneMeasurementFourChunksWithDuplicates {}
 #[async_trait]
 impl DbSetup for OneMeasurementFourChunksWithDuplicates {
     async fn make(&self) -> Vec<DbScenario> {
-        let db = make_db().await.db;
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/library.rs:758]: This function include OG and NG scenarios. Remove the OG ones

File: query_tests/src/scenarios/library.rs [hunk 11]
```diff
@@ -810,109 +656,35 @@ impl DbSetup for OneMeasurementFourChunksWithDuplicates {
...
+                ..Default::default()
+            },
+        ])
+        .await;
 
-#[derive(Debug)]
-/// Setup for four chunks with duplicates for deduplicate tests
-///
-/// This scenario is OG-specific and can be used for `EXPLAIN` plans and system tables.
-pub struct OldOneMeasurementFourChunksWithDuplicates {}
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/library.rs:857]: This was ported

File: query_tests/src/scenarios/library.rs [hunk 12]
```diff
@@ -121,6 +41,7 @@ impl DbSetup for MeasurementWithMaxTime {
     }
 }
 
+// TODO: rewrite this for NG. Need to see whether we will need to test empty DB and if so how
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/library.rs:44]: #4488

File: query_tests/src/scenarios/library.rs [hunk 13]
```diff
@@ -810,109 +656,35 @@ impl DbSetup for OneMeasurementFourChunksWithDuplicates {
...
         scenarios
     }
 }
 
-#[derive(Debug)]
-/// This has a single scenario with all the life cycle operations to
-/// test queries that depend on that
-///
-/// This scenario is OG-specific and can be used for `EXPLAIN` plans and system tables.
-pub struct OldTwoMeasurementsManyFieldsLifecycle {}
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/library.rs:877]: This is no longer relevant

File: query_tests/src/scenarios/util.rs [hunk 14]
```diff
@@ -537,242 +407,6 @@ async fn make_chunk_with_deletes_at_different_stages(
     }
 }
 
-async fn make_chunk_with_deletes_at_different_stages_old(
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/util.rs:540]: This was ported

File: query_tests/src/scenarios/util.rs [hunk 15]
```diff
@@ -795,205 +429,6 @@ async fn make_chunk_with_deletes_at_different_stages_new(
     DbScenario { scenario_name, db }
 }
 
-/// Build many chunks which are in different stages
-//  Note that, after a lot of thoughts, I decided to have 2 separated functions, this one and the one above.
-//  The above tests delete predicates before and/or after a chunk is moved to different stages, while
-//  this function tests different-stage chunks in various stages when one or many deletes happen.
-//  Even though these 2 functions have some overlapped code, merging them in one
-//  function will created a much more complicated cases to handle
-pub async fn make_different_stage_chunks_with_deletes_scenario_old(
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/util.rs:804]: This was ported

File: query_tests/src/scenarios/util.rs [hunk 16]
```diff
@@ -1005,147 +440,7 @@ pub async fn make_two_chunk_scenarios(
     data1: &str,
     data2: &str,
 ) -> Vec<DbScenario> {
-    make_two_chunk_scenarios_old(partition_key, data1, data2)
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/util.rs:1008]: was ported

File: query_tests/src/scenarios/util.rs [hunk 17]
```diff
@@ -1221,45 +516,6 @@ pub async fn make_n_chunks_scenario_new(chunks: &[ChunkDataNew<'_, '_>]) -> Vec<
     scenarios
 }
 
-// // This function loads one chunk of lp data into RUB for testing predicate pushdown
-pub(crate) async fn make_one_rub_or_parquet_chunk_scenario(
```
Comment (NGA-TRAN) [inline query_tests/src/scenarios/util.rs:1225]: no longer relevant

File: query_tests/src/sql.rs [hunk 18]
```diff
@@ -123,28 +123,6 @@ async fn sql_select_from_school() {
...
-        "| 0  | Success | true       | true          | true           | h2o        | 1970-01-01T00 | Compacting chunks to ReadBuffer     |",
-        "| 1  | Success | true       | true          | true           | h2o        | 1970-01-01T00 | Persisting chunks to object storage |",
-        "| 2  | Success | true       | true          | true           | h2o        | 1970-01-01T00 | Writing chunk to Object Storage     |",
-        "+----+---------+------------+---------------+----------------+------------+---------------+-------------------------------------+",
-    ];
-
-    // Check that the cpu time used reported is greater than zero as it isn't
-    // repeatable
-    run_sql_test_case(
-        OldTwoMeasurementsManyFieldsLifecycle {},
```
Comment (NGA-TRAN) [inline query_tests/src/sql.rs:142]: no longer relevant

File: query_tests/src/scenarios.rs [hunk 19]
```diff
@@ -52,12 +52,8 @@ pub fn get_all_setups() -> &'static HashMap<String, Arc<dyn DbSetup>> {
             register_setup!(TwoMeasurementsManyFields),
             register_setup!(TwoMeasurementsPredicatePushDown),
             register_setup!(TwoMeasurementsManyFieldsOneChunk),
-            register_setup!(OldTwoMeasurementsManyFieldsTwoChunks),
```
Comment (alamb) [inline query_tests/src/scenarios.rs:55]: The fact that the tests that were OG specific were named `OldXXX` make reviewing this PR very easy. Thanks @NGA-TRAN  and @crepererum

File: query_tests/src/scenarios/delete.rs (hunk 3 above)
Comment (alamb) [reply query_tests/src/scenarios/delete.rs:258]: FWIW I found the whitespace blind diff https://github.com/influxdata/influxdb_iox/pull/4487/files?w=1  easer to see what had changed

File: query_tests/src/scenarios/delete.rs [hunk 20]
```diff
@@ -255,226 +251,76 @@ impl DbSetup for ThreeDeleteThreeChunks {
...
+                partition_key,
+                ..Default::default()
             },
-            ChunkDataOld {
-                lp_lines: lp_lines_2.clone(),
-                chunk_stage: ChunkStageOld::Os,
+            ChunkDataNew {
+                lp_lines: lp_lines_2,
+                preds: preds.clone(),
+                // preds: vec![
```
Comment (alamb) [inline query_tests/src/scenarios/delete.rs:296]: given that `preds` is the same, I don't understand why this item (`lp_lines_2) has two commented out `PredNew`s but the previous one (`lp_lines_1) has three commented out `PredNew`s.

Is it because some of the predicates don't affect the other `lp_line`s? Maybe a clarifying comment could help

File: query_tests/src/scenarios/library.rs (hunk 7 above)
Comment (alamb) [reply query_tests/src/scenarios/library.rs:24]: I am not sure that `query_tests` is the best place to test NG lifecycle 

I always envisioned `query_tests` being used to ensure query results were consistent with all the various physical layouts (e.g. different chunk / orders / arrangements) rather than testing the creation of those physical layouts itself

File: query_tests/src/scenarios/library.rs (hunk 12 above)
Comment (alamb) [reply query_tests/src/scenarios/library.rs:44]:
```suggestion
// TODO: rewrite this for NG. Need to see whether we will need to test empty DB and if so how
// <https://github.com/influxdata/influxdb_iox/issues/4488>
```

File: query_tests/src/scenarios/util.rs [hunk 21]
```diff
@@ -45,14 +38,6 @@ use std::{fmt::Display, sync::Arc};
 // & when delete predicates are applied
 
 // STRUCTs & ENUMs
-#[derive(Debug, Clone)]
```
Comment (alamb) [inline query_tests/src/scenarios/util.rs:48]: 🎉

**Review by alamb — APPROVED** [review]
Looks great to me. Thanks @NGA-TRAN 

Maybe a good follow on PR would be to remove all the `New` in the names (as we no longer have dual `New` and `Old`)

File: query_tests/src/scenarios/delete.rs (hunk 20 above)
Comment (NGA-TRAN) [reply query_tests/src/scenarios/delete.rs:296]: The original OG test wanted to apply all 3 delete predicates on each chunk to see if the deletes are applied correctly even if they have no effects. The test was ported to NG and modified a bit to only apply predicates that have real effects. I modify it back to original intention. I have added a comment on top to explain the purpose. I have also removed this comments

File: query_tests/src/scenarios/library.rs (hunk 7 above)
Comment (NGA-TRAN) [reply query_tests/src/scenarios/library.rs:24]: Sounds good. I have removed the TODO

**Comment by NGA-TRAN** [comment]
@alamb 
> Maybe a good follow on PR would be to remove all the New in the names (as we no longer have dual New and Old)

Yes, that was the plan

**Comment by alamb** [comment]
> Yes, that was the plan

Nice -- I didn't realize that
